explorer/testdata/class/class_inheritance_shadow.carbon

```diff
@@ -0,0 +1,42 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: %{explorer} %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes=false %s
+// RUN: %{explorer} --parser_debug --trace_file=- %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes %s
+// AUTOUPDATE: %{explorer} %s
+// CHECK: Class D
+// CHECK: d.value = 2
+// CHECK: d.GetValue() = 2
+// CHECK: result: 0
+
+package ExplorerTest api;
+
+base class C {
+  fn Method1() {
+    Print("Class C");
+  }
+  fn GetValue[me: Self]() -> i32 {
+    return me.value;
+  }
+  var value: i32;
+}
+
+class D extends C {
+  fn Method1() {
+    Print("Class D");
+  }
+  var value: i32;
+}
+
+fn Main() -> i32 {
+  // Initialize derived value first, base value second
+  var d: D = {.value = 2, .value = 1};
```

Comment:
The current way to assign base class values when shadowed seems wrong to me. The [class design doc](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/classes.md#constructors) discusses the use of `{.base = ...` for base class initialization in constructors, but this isn't supported yet AFAIS.

I couldn't find a mention of this particular situation in the design.

This example still illustrates variable shadowing, and the fact that `me` represents the object. and not the base class, but we might likely want to not allow that in some future work

Reply:
This would probably need to have support added as part of `extends` but is probably the right direction. I suspect this example actually shouldn't work -- the shadowing should be an ambiguous name lookup (I think this is the direction that function calls will go too).

Note, something that might be okay/supportable would be: `{.(C.value) = 2, .(D.value) = 1};`

But that kind of name specification is also how ambiguous functions can be resolved.

Reply:
Agree about the initialization.

Regarding the `base =`, are you saying that something that you'd want to include in this PR, or changing that test to not use this form?